servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/BaseHostSelector.java

```diff
@@ -67,21 +68,38 @@ protected final Single<C> noActiveHostsFailure(List<Host<ResolvedAddress, C>> us
                 this.getClass(), "selectConnection(...)"));
     }
 
+    // TODO: this could really be the core method on `Host` other than the nullable part. We could use Optional...
+    //  The API of passing in the status is weird but we could either live with it or just ask the question again
+    //  if we need to make a new connection.
+    protected final @Nullable Single<C> selectFromHost(
```

Comment:
agreed, the status is indeed weird as a parameter.

Reply:
I'll just remove it. It's only used for new connections anyway.